sdk/python/kfp/compiler/compiler_test.py

```diff
@@ -792,6 +792,55 @@ def my_pipeline() -> NamedTuple('Outputs', [
             ]):
                 task = print_op(msg='Hello')
 
+    def test_nested_conditions_with_for_loops_in_nested_pipelines(self):
+
+        @dsl.pipeline()
+        def pipeline_2():
+            with dsl.ParallelFor([1, 2, 3]):
+                print_op(message='1')
+
+        @dsl.pipeline()
+        def pipeline_3(run_pipeline_2: int = 1):
+            with dsl.ParallelFor([1, 2, 3]):
+                print_op(message='1')
+
+                with dsl.Condition(run_pipeline_2 == run_pipeline_2):
+                    with dsl.Condition(run_pipeline_2 == 1):
+                        pipeline_2()
+
+        @dsl.pipeline()
+        def pipeline_1(run_pipeline_2: int = 1, run_pipeline_3: int = 1):
+            with dsl.Condition(run_pipeline_2 == run_pipeline_2):
+                with dsl.Condition(run_pipeline_2 == 1):
+                    pipeline_2()
+
+            with dsl.Condition(run_pipeline_3 == run_pipeline_3):
+                with dsl.Condition(run_pipeline_3 == 1):
+                    pipeline_3()
```

Comment:
Is this a minimal example? If not, could we reduce this to the minimal example required to reproduce the problematic behavior? This will help us ensure we don't regress and will reduce the likelihood that the test is removed as redundant or unclear in the future.

Also, consider renaming the test method to reflect the behavior under test and, if needed, adding a docstring to describe in more detail.

Reply:
Should be much better now, thanks for the push